### PR TITLE
Level Generation Editorial

### PR DESCRIPTION
**Pull request by eysbutno**
_Place an "x" in the corresponding checkbox if it is done or does not apply to this pull request._

- [x] I have tested my code.
- [x] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [x] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions).
  - I understand that if it is clear that I have not attempted to follow these conventions, my PR will be closed.
  - If changes are requested, I will re-request a review after addressing them.
- [x] I have linked this PR to any issues that it closes.

AC: https://codeforces.com/contest/818/submission/295323118

### CODE REVIEW

File: solutions/silver/cf-818F.mdx [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+
+<Spoiler title="Hint 1">
+
+What is the upper bound for the number of bridges we can have?
+
+</Spoiler>
+
+<Spoiler title="Answer to Hint 1">
+
+The optimal construction would be to create a tree, which has $n - 1$ bridges.
```
Comment (SansPapyrus683) [inline solutions/silver/cf-818F.mdx:16]: 1. a direct answer would probably be better
2. you can't just say that :sob:

File: solutions/silver/cf-818F.mdx [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+
+int main() {
+	int test_num;
+	cin >> test_num;
+	for (int i = 0; i < test_num; i++) {
+		int nodes;
+		cin >> nodes;
+
+		/** @return if we can make a graph with this many edges */
+		const auto check = [&](ll edges) -> bool {
```
Comment (SansPapyrus683) [inline solutions/silver/cf-818F.mdx:63]: you only call this once, just put it directly in the bin search

File: solutions/silver/cf-818F.mdx (hunk 1 above)
Comment (TheGamingMousse) [reply solutions/silver/cf-818F.mdx:16]: wait wdym by "you can't just say that" lol

File: solutions/silver/cf-818F.mdx (hunk 2 above)
Comment (TheGamingMousse) [reply solutions/silver/cf-818F.mdx:63]: fair enough

File: solutions/silver/cf-818F.mdx (hunk 1 above)
Comment (SansPapyrus683) [reply solutions/silver/cf-818F.mdx:16]: surely at least give some intuition as to why it's the case

Comment (TheGamingMousse) [reply solutions/silver/cf-818F.mdx:16]: hmm im not really sure how to lead into the idea of forming a tree without explicitly saying it

Comment (SansPapyrus683) [reply solutions/silver/cf-818F.mdx:16]: well i guess that's the hard part of writing an editorial

Comment (TheGamingMousse) [reply solutions/silver/cf-818F.mdx:16]: true lol

ngl i think just directly saying it might be more useful

File: solutions/silver/cf-818F.mdx [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+Let's consider allocating a certain amount of these nodes to forming a tree.
+How can we use the rest of these nodes to use as many edges as possible?
+
+</Spoiler>
+
+<Spoiler title="Explanation">
+
+The most bridges we can have in a graph with $n$ nodes is $n - 1$ bridges. For
+those interested, the reason this is the case is that all of our bridges must be part of an
+arbitrary spanning tree of our graph.
```
Comment (ryanchou-dev) [inline solutions/silver/cf-818F.mdx:26]: move this to hint 1 - also silver competitors probably don't know what a spanning tree is? you should clarify this 👍

File: solutions/silver/cf-818F.mdx [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+```cpp
+#include <bits/stdc++.h>
+using namespace std;
+
+using ll = long long;
+
+int main() {
+	int test_num;
+	cin >> test_num;
+	for (int i = 0; i < test_num; i++) {
```
Comment (ryanchou-dev) [inline solutions/silver/cf-818F.mdx:53]: nitpick but make `i` -> `_`

**Review by ryanchou-dev — COMMENTED** [review]
cooked 👨‍🍳

File: solutions/silver/cf-818F.mdx (hunk 4 above)
Comment (SansPapyrus683) [reply solutions/silver/cf-818F.mdx:53]: uh i would disagree but up to you lmao

Comment (TheGamingMousse) [reply solutions/silver/cf-818F.mdx:53]: yeah i might make it like `t` or something but `_` feels a bit weird in C++ tbh

File: solutions/silver/cf-818F.mdx (hunk 3 above)
Comment (TheGamingMousse) [reply solutions/silver/cf-818F.mdx]: done, lmk if its still confusing/misplaced

File: solutions/silver/cf-818F.mdx [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+source: CF
+title: Level Generation
+author: Justin Ji
+---
+
+<Spoiler title="Hint 1">
+
+The upper bound for the number of bridges we can have is $n - 1$ because
+each bridge is present in any spanning tree of the graph. A
+spanning tree is a subgraph that connects every node without cycles.
```
Comment (SansPapyrus683) [inline solutions/silver/cf-818F.mdx:12]: maybe just link to the wikipedia or brilliant defn

**Review by SansPapyrus683 — APPROVED** [review]

File: solutions/silver/cf-818F.mdx (hunk 5 above)
Comment (TheGamingMousse) [reply solutions/silver/cf-818F.mdx:12]: good point

**Comment by TheGamingMousse** [comment]
oops - before cryan beats me to it i gotta add the official edi lol

**Review by ryanchou-dev — APPROVED** [review]

**Pull request merged**